QuickOSM/core/process.py

```diff
@@ -150,13 +151,18 @@ def open_file(
             if config_outputs and config_outputs[layer]['style']:
                 new_layer.loadNamedStyle(config_outputs[layer]['style'])
             else:
-                # Loading default styles
-                if layer == "multilinestrings" or layer == "lines":
-                    if "colour" in item['tags']:
-                        new_layer.loadNamedStyle(
-                            join(dirname(dirname(abspath(__file__))),
-                                 "styles",
-                                 layer + "_colour.qml"))
+                if "colour" in item['tags']:
+                    colors = []
+                    for feature in new_layer.getFeatures():
+                        colors.append(feature.attribute('colour'))
```

Comment:
This won't be efficient because te `getFeatures` will retrieve a lot of information (all fields, all geometries) etc.
Use QgsFeatureRequest to optimize **a lot**.
Read https://nyalldawson.net/2016/10/speeding-up-your-pyqgis-scripts/

But anyway, in your case, use `QgsVectorLayer.uniqueValues`